src/util/prop-definition.js

```diff
@@ -0,0 +1,104 @@
+import dashCase from './dash-case';
+
+/**
+ * @internal
+ * Property Definition
+ *
+ * Internal meta data and strategies for a property.
+ * Created from the options of a PropOptions config object.
+ *
+ * Once created a PropDefinition should be treated as immutable and final.
+ * 'getPropsMap' function memoizes PropDefinitions by Component's Class.
+ *
+ * The 'attribute' option is normalized into the 'attrName' property.
+ */
+export default class PropDefinition {
+
+  // constructor(name:string|symbol, cfg:PropOptions) {
+  constructor (nameOrSymbol, cfg) {
+    this._name = nameOrSymbol;
+
+    cfg = cfg || {};
+
+    if (typeof cfg === 'function') {
+      // todo: Where is documented that a config can just be the coerce function?
+      cfg = {coerce: cfg};
+    }
+
+    // Set Default values
+
+    this.attrName = null;
+
+    this.coerce = null;
+
+    // todo: we probabbly need to update the doc
+    // from doc one would think default value is undefined
+    // value was defined inside props-init.js
+    this.default = null;
+
+    // todo: should be JSON.stringify ?
```

Comment:
Not by default. As mentioned below, this would be unnecessarily expensive and opinionated.